_episodes/01-introduction.md

```diff
@@ -63,4 +89,30 @@ However, we still need a way to tell the generator how we want our content
 to look when it's displayed in the browser.
 For that, we will use a tool called Markdown.
 
+> ## The perfect tool for the job
+>
+> Given the following types of websites,
+> reason if a static site generator is an appropriate solution to implement them.
+>
+> 1. A personal website with *About* and *Projects* sections
+> 2. A forum or discussion platform
+> 3. A community blog or news website
+> 4. A search engine (such as google.com)
+> 5. A wiki (such as wikipedia.com)
+> 6. An online book
+>
+> > ## Solution
+> >
+> > 1. In most cases, **Yes**. This kind of content is typically written/edited by one person and meant to have a read-only access to visitors.
+> > 2. Most likely **No**. Such website requires interactivity and ways to identify who wrote what content.
+> >
+> > For questions 3 to 5 the answer is both **Yes** and **No** depending on the requirements and necessary functionality.
+> >
+> > 3. A simple blog or news website, maintained by a small set of users, is perfectly achievable by using a static generator. For very large groups of content creators or if access to articles needs to be controlled individually, using a static generator will lead to difficult technical challenges.
+> > 4. Implementing something as sophisticated as Google's search would be close to impossible with a static generator. That said, it's possible to have a simple engine that searches across all pages produced by a static generator using clever indexing solutions.
```

Comment:
I think the answer to no. 4 should be left at "no." If you're using Google as an example, then it's reasonable for the learner to conclude that an SSG is not the appropriate tool for the job. If you can find a link to a suitable tutorial/understandable example, you could give this answer:

> No. Searching large volumes of pages would require communication with an index stored on a server. (However, [solutions do exist] (URL) to make your static site searchable.)

Reply:
The example I know of is the search engine in sphinx. E.g. [Python docs search](https://docs.python.org/3/search.html?q=os&check_keywords=yes&area=default).
But I'm happy with leaving it at No

Reply:
Rephrased to go as No.